packaging/language/pear.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2014, Leszek Krupiński <leafnode@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import re
+
+DOCUMENTATION = '''
+---
+module: pear
+short_description: Manages PHP PEAR library dependencies.
+description:
+     - "Manage PHP PEAR dependencies. To use this module, one of the following keys is required: C(name)
+       or C(channel)."
+version_added: "0.1"
+options:
+  name:
+    description:
+      - The name of a PEAR library to install.
+    required: false
+    default: null
+  version:
+    description:
+      - Version of a package to install. If version is omitted, newest version of a package will be installed.
+    required: false
+    default: null
+  channel:
+    description:
+      - URL of a PEAR channel to add.
+    required: false
+    default: null
+  state:
+    description:
+      - Desired state of a package or channel.
+    required: false
+    default: present
+    choices: [ "present", "absent" ]
+notes:
+   - PEAR package manager has to be installed on a remote host. 
+requirements: [ "pear" ]
+author: Leszek Krupiński
+'''
+
+EXAMPLES = '''
+# Install Log PEAR package
+- pear: name=Log
+
+# Uninstall Net_DNS2 package
+- pear: name=Net_DNS2 state=absent
+
+# Add pear.symfony.com PEAR channel
+- pear: channel=pear.symfony.com
+
+# Remove pear.symfony.com PEAR channel
+- pear: channel=pear.symfony.com state=absent
+'''
+
+def _get_full_name(name, version=None):
+    if version is None:
+        resp = name
+    else:
+        resp = name + '-' + version
+    return resp
+
+def _is_package_present(module, name, version):
+    rc, out_pear, err_pear = module.run_command('pear list')
```

Comment:
don't rely on path, use module's find bin to locate the correct pear executable (same in other instances below)